server/src/test/java/org/opensearch/index/ShardIndexingPressureConcurrentExecutionTests.java

```diff
@@ -71,15 +76,11 @@ public void testCoordinatingPrimaryThreadedUpdateToShardLimits() throws Exceptio
             NUM_THREADS * 15,
             shardIndexingPressure.shardStats().getIndexingPressureShardStats(shardId1).getCurrentCombinedCoordinatingAndPrimaryBytes()
         );
-        assertTrue(
-            (double) (NUM_THREADS * 15) / shardIndexingPressure.shardStats()
-                .getIndexingPressureShardStats(shardId1)
-                .getCurrentPrimaryAndCoordinatingLimits() < 0.95
-        );
-        assertTrue(
+        MatcherAssert.assertThat(
             (double) (NUM_THREADS * 15) / shardIndexingPressure.shardStats()
                 .getIndexingPressureShardStats(shardId1)
-                .getCurrentPrimaryAndCoordinatingLimits() > 0.75
+                .getCurrentPrimaryAndCoordinatingLimits(),
+            allOf(greaterThan(0.75), lessThanOrEqualTo(0.95))
```

Comment:
@getsaurabh02 To be honest I don't fully understand this assertion. Can you verify it is safe/correct to change this assertion from "less than 0.95" to "less than or equal to 0.95"?

Reply:
The additional memory allocation per shard is governed by its peak memory utilization threshold of the current allocation. This is configurable setting and defaults to [95%](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/index/ShardIndexingPressureMemoryManager.java#L78-L84). 

Once the current allocation is greater than this threshold, the new shard limits are calculated and allocated. Since [this](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/index/ShardIndexingPressureMemoryManager.java#L445) condition checks for current value to be greater than, it should be safe to assume that the current limit can reach until the 95% (0.95) utilization. Hence "less than or equal to 0.95" should be good.